Assignment6/Array.cs

```diff
@@ -1,10 +1,84 @@
 ﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 
 namespace Assignment6
 {
-    public class Array<T>
+    public class Array<T> : System.Collections.Generic.ICollection<T>

```

Comment:
```suggestion
    public class Array<T> : ICollection<T>
```

You don't need the whole path here.